controller/dataplane/controller_reconciler_utils.go

```diff
@@ -77,6 +84,170 @@ func (r *Reconciler) ensureDataPlaneAddressesStatus(
+			if err := c.Delete(ctx, &cm); client.IgnoreNotFound(err) != nil {
+				return nil, ctrl.Result{}, err
+			}
+		}
+	}
+
+	return cps, ctrl.Result{}, nil
+}
+
+func findCustomPluginConfigMapsOwnedByDataPlane(
+	ctx context.Context, c client.Client, dataplane *operatorv1beta1.DataPlane,
+) ([]corev1.ConfigMap, error) {
+	cms, err := k8sutils.ListConfigMapsForOwner(ctx, c, dataplane.GetUID())
+	if err != nil {
+		return nil, err
+	}
+	return lo.Filter(cms, func(cm corev1.ConfigMap, _ int) bool {
+		_, isForKPI := cm.Annotations[consts.AnnotationMappedToKongPluginInstallation]
+		return isForKPI
+	}), nil
+}
+
+func populateDedicatedConfigMapForKongPluginInstallation(
+	ctx context.Context,
+	logger logr.Logger,
+	c client.Client,
+	cms []corev1.ConfigMap,
+	kpiNN types.NamespacedName,
+	dataplane *operatorv1beta1.DataPlane,
+) (customPlugin, ctrl.Result, error) {
+	kpi, ready, err := verifyKPIReadinessForDataPlane(ctx, logger, c, dataplane, kpiNN)
+	if err != nil {
+		return customPlugin{}, ctrl.Result{}, err
+	}
+	if !ready {
+		return customPlugin{}, ctrl.Result{Requeue: true}, nil
+	}
+
+	var underlyingCM corev1.ConfigMap
+	backingCMNN := types.NamespacedName{
+		Namespace: kpi.Namespace,
+		Name:      kpi.Status.UnderlyingConfigMapName,
+	}
+	log.Trace(logger, fmt.Sprintf("Fetch underlying ConfigMap %s for KongPluginInstallation", backingCMNN), kpi)
+	if err := c.Get(ctx, backingCMNN, &underlyingCM); err != nil {
+		return customPlugin{}, ctrl.Result{}, fmt.Errorf("could not fetch underlying ConfigMap to clone %s: %w", backingCMNN, err)
+	}
+
+	log.Trace(logger, "Find ConfigMap mapped to KongPluginInstallation", kpi)
+	mappedConfigMapForKPI := lo.Filter(cms, func(cm corev1.ConfigMap, _ int) bool {
+		kpiNN := cm.Annotations[consts.AnnotationMappedToKongPluginInstallation]
+		return kpiNN == client.ObjectKeyFromObject(&kpi).String()
+	})
+	var cm corev1.ConfigMap
+	switch len(mappedConfigMapForKPI) {
+	case 0:
+		log.Trace(logger, "Create new ConfigMap for KongPluginInstallation", kpi)
+		cm.GenerateName = dataplane.Name + "-"
+		cm.Namespace = dataplane.Namespace
+		k8sutils.SetOwnerForObject(&cm, dataplane)
+		k8sresources.LabelObjectAsDataPlaneManaged(&cm)
+		k8sresources.AnnotateConfigMapWithKongPluginInstallation(&cm, kpi)
+		cm.Data = underlyingCM.Data
+		if err := c.Create(ctx, &cm); err != nil {
+			return customPlugin{}, ctrl.Result{}, fmt.Errorf("could not create new ConfigMap for KongPluginInstallation: %w", err)
+		}
+	case 1:
+		cm = mappedConfigMapForKPI[0]
+		log.Trace(logger, fmt.Sprintf("Update existing ConfigMap %s for KongPluginInstallation", client.ObjectKeyFromObject(&cm)), kpi)
+		cm.Data = underlyingCM.Data
+		if err := c.Update(ctx, &cm); err != nil {
+			if k8serrors.IsConflict(err) {
+				return customPlugin{}, ctrl.Result{Requeue: true}, nil
+			}
+			return customPlugin{}, ctrl.Result{}, fmt.Errorf("could not update mapped: %w", err)
+		}
```

Comment:
While this seems OK I'd suggest considering following the pattern that we already have, that is comparing relevant fields and only if those are not as expected, performing an update. This could potentially save some unnecessary updates.

Not critical though to be included in this PR.